test/unit/argument_parser/argument_parser_design_error_test.cpp

```diff
@@ -11,6 +11,19 @@
 
 using namespace seqan3;
 
+TEST(parser_design_error, app_name_validation)
+{
+    const char * argv[] = {"./argument_parser_test"};
+
+    EXPECT_NO_THROW((argument_parser parser{"test_parser", 1, argv}));
```

Comment:
I think you should do this:

```suggestion
    EXPECT_NO_THROW((argument_parser{"test_parser", 1, argv}));
```

Reply:
Or
```suggestion
    EXPECT_NO_THROW(([]()
    {
        argument_parser parser{"test_parser", 1, argv};
    }));
```